google-cloud-tools-plugin/ultimate/src/com/google/cloud/tools/intellij/appengine/server/instance/AppEngineRunConfigurationEditor.java

```diff
@@ -114,11 +114,6 @@ protected void resetEditorFrom(CommonModel commonModel) {
     final AppEngineServerModel serverModel = (AppEngineServerModel) commonModel.getServerModel();
     final Artifact artifact = serverModel.getArtifact();
     myArtifactComboBox.setSelectedItem(artifact);
-    if (artifact == null && myArtifactComboBox.getItemCount() == 1) {
```

Comment:
Indeed... I don't think we have enough information at this point to distinguish a Java project from a Java EE one.